ui/admin/app/routes/scopes/scope/roles/role/manage-scopes/manage-custom-scopes.js

```diff
@@ -40,27 +40,96 @@ export default class ScopesScopeRolesRoleManageScopesManageCustomScopesRoute ext
       scope_id: [{ equals: scope_id }],
     };
 
-    const subScopes = await this.store.query('scope', {
+    const orgScopes = await this.store.query('scope', {
       scope_id,
       query: { search, filters },
       page,
       pageSize,
     });
-    const totalItems = subScopes.meta?.totalItems;
+    const totalItems = orgScopes.meta?.totalItems;
     const totalItemsCount = await this.getTotalItemsCount(
       scope_id,
       search,
       totalItems,
     );
 
+    const projectsSelected = await this.getProjectsSelected(
+      role.grantScopeProjectIDs,
+      orgScopes,
+    );
+
     return {
       role,
-      subScopes,
+      orgScopes,
+      projectsSelected,
       totalItems,
       totalItemsCount,
     };
   }
 
+  async getProjectsSelected(projectIDs, orgScopes) {
+    const options = { pushToStore: false, peekIndexedDB: true };
+    const projectsSelected = {};
+
+    const projectScopes = await Promise.all(
```

Comment:
I'm just wondering, was it necessary to query for each individual project scope or could you have gathered all the IDs up and then just did one query?

And same with the `selectedProjects`

Reply:
the way im thinking about this is that a separate query(to peek the DB)  is made for each org scope to determine the number of projects that exist under the particular org. And for `selectedProjects` the query is being narrowed to projects that are also grant scope ids for a particular org. This entire getter is used for the "Projects Selected" column in the table that gets project info for each org in the "manage custom scopes" page.

Reply:
Updated this to be just one query, thank you!